diesel/src/pg/expression/expression_methods.rs

```diff
@@ -3749,4 +3750,16 @@ pub(in crate::pg) mod private {
     impl TextArrayOrNullableTextArray for Array<Nullable<Text>> {}
     impl TextArrayOrNullableTextArray for Nullable<Array<Text>> {}
     impl TextArrayOrNullableTextArray for Nullable<Array<Nullable<Text>>> {}
+
+    #[diagnostic::on_unimplemented(
+        message = "`{Self}` is neither `Record<Text>`, `Record<Nullable<Text>>`,\
+                   `Nullable<Record<Text>>` nor `diesel::sql_types::Nullable<Record<Nullable<Text>>>`",
+        note = "try to provide an expression that produces one of the expected sql types"
+    )]
+    pub trait RecordOrNullableRecord {}
+
+    impl RecordOrNullableRecord for Record<Text> {}
+    impl RecordOrNullableRecord for Record<Nullable<Text>> {}
+    impl RecordOrNullableRecord for Nullable<Record<Text>> {}
+    impl RecordOrNullableRecord for Nullable<Record<Nullable<Text>>> {}
```

Comment:
`Record` is a somewhat especial type, in so far that it always accepts a tuple as generic argument. So something like `Record<(Integer, Text)>` or `Record<(Text, Date, Integer)>`. 

The right variant here would be to write something like `impl<T> RecordOrNullableRecord for Record<T> {}` and `impl<T> RecordOrNullableRecord for Nullable<Record<T>> {}`

Reply:
Thanks @weiznich , I'll make these changes